bin/crowbar_batch

```diff
@@ -329,6 +333,20 @@ def merge_attributes(new_json, barclamp, proposal)
   Chef::Mixin::DeepMerge.deep_merge!(to_merge, new_json)
 end
 
+def prevent_password_lockout(attrs)
+  return unless attrs
+
+  users = attrs['users']
+  return unless users && users[@username]
+
+  if @password != users[@username]['password']
+    warn "Refusing to change password for '#{@username}' user " + \
+         #"from #{@password} to #{users[@username]['password']} " + \
+         "which would lock myself out!"
+  end
+  to_merge['attributes'][barclamp]['users'].delete @username
```

Comment:
While testing this I got a 
```
/opt/dell/bin/barclamp_lib.rb:536:in `eval': undefined local variable or method `to_merge' for main:Object (NameError)
```
Changing it to an instance variable `@to_merge` (in all its lines ofc) fixes this.

Second issue in this line: `barclamp` is unknown as-well. Making it a method argument fixed it for me.